mlflow/utils/search_utils.py

```diff
@@ -355,6 +369,31 @@ def _get_value(cls, identifier_type, key, token):
                     "Got value {value}".format(value=token.value),
                     error_code=INVALID_PARAMETER_VALUE,
                 )
+        elif identifier_type == cls._DATASET_IDENTIFIER:
+            if key in cls.NUMERIC_ATTRIBUTES:
+                if token.ttype not in cls.NUMERIC_VALUE_TYPES:
+                    raise MlflowException(
+                        "Expected numeric value type for numeric attribute: {}. "
+                        "Found {}".format(key, token.value),
+                        error_code=INVALID_PARAMETER_VALUE,
+                    )
+                return token.value
+            elif token.ttype in cls.STRING_VALUE_TYPES or isinstance(token, Identifier):
+                return cls._strip_quotes(token.value, expect_quoted_value=True)
```

Comment:
Are we validating the attribute keys that are compatible with `datasets`? I.e. if I do `datasets.foobazbiz = 'bar'`, is `foobazbiz` treated as valid at this layer? I think it should throw.